### DISCUSSION TITLE
sd.next - installed today - FileNotFoundError: [Errno 2] No such file or directory: - but directory is there ?

### DISCUSSION THREAD

**Discussion by cgidesign-de** [vladmandic/sdnext#2404] [Q&A]
New install, new issue:

app: SD.next
updated: 2023-10-24
on Windows 10 with RTX 3090.

Text 2 image:
It creates the images and it stores them in the respective output dir as expected.

But, even though the images are in the output, I get the below error, when the generation sequence finishes (e.g. if batch size is 3, it happens after the third image). I have to restart sd.next cmd after each generation sequence now.

I have one difference to a default windows. The Windows temp dir is not at its default location but on a different drive:
D:\Windows\User\home\Temp\

This did not cause issues with sd.next in the past but it does now with the latest install.

Any idea?

Traceback (most recent call last):
  File "F:\automatic\venv\lib\site-packages\gradio\queueing.py", line 388, in call_prediction
    output = await route_utils.call_process_api(
  File "F:\automatic\venv\lib\site-packages\gradio\route_utils.py", line 219, in call_process_api
    output = await app.get_blocks().process_api(
  File "F:\automatic\venv\lib\site-packages\gradio\blocks.py", line 1440, in process_api
    data = self.postprocess_data(fn_index, result["prediction"], state)
  File "F:\automatic\venv\lib\site-packages\gradio\blocks.py", line 1341, in postprocess_data
    prediction_value = block.postprocess(prediction_value)
  File "F:\automatic\venv\lib\site-packages\gradio\components\gallery.py", line 196, in postprocess
    file = self.pil_to_temp_file(img, dir=self.DEFAULT_TEMP_DIR)
  File "F:\automatic\modules\ui_tempdir.py", line 65, in pil_to_temp_file
    with tempfile.NamedTemporaryFile(delete=False, suffix=".png", dir=dir) as tmp:
  File "C:\Program Files\Python310\lib\tempfile.py", line 559, in NamedTemporaryFile
    file = _io.open(dir, mode, buffering=buffering,
  File "C:\Program Files\Python310\lib\tempfile.py", line 556, in opener
    fd, name = _mkstemp_inner(dir, prefix, suffix, flags, output_type)
  File "C:\Program Files\Python310\lib\tempfile.py", line 256, in _mkstemp_inner
    fd = _os.open(file, flags, 0o600)
FileNotFoundError: [Errno 2] No such file or directory: 'D:\\Windows\\User\\home\\Temp\\gradio\\tmph_8hhhx8.png'

**Comment by cgidesign-de** [comment]
Update:
I manually added the subdir "gradio".
Now it is working again.

**Comment by vladmandic** [comment]
the fact that error happens at the end of the batch pretty much indicates that it happens during saving of grid.
and if you have grid auto-save enabled, it would save it to grid location, not windows temp file.

regarding temp folder, can you reproduce with DEFAULT windows temp location? 
otherwise this most likely looks like a permission issue.

**Comment by cgidesign-de** [comment]
"always save all generated grids" is off
"grid row count" is -1

Might be rights: I just checked that "users" group does not have write / modify right - Administrators though have.
Seems so sd is running within users context even though my local account belongs to admins.

Anyway, I consider this solved.

**Reply by vladmandic** [reply]
> Seems so sd is running within users context even though my local account belongs to admins.

that is how windows works pretty much always - it reduces priviledges unless explicit priviledge is actually requested.
so if user is part of admin group means you're allowed to ask for admin priviledges, but by default task will run as user.
